### ISSUE TITLE
Every subsequent call to machine.interfaces takes longer

### ISSUE THREAD

**Issue by kmadac** [maas/python-libmaas#210]
Calling of `machine.interfaces` several times takes working with interface object longer and longer. After 8 calls, script is freezed. I experienced same issue when calling `len(interface.links)`.

```
for i in range(0,10):
    start = time.time()
    interface = machine.interfaces.get_by_name('bond0')
    print(len(interface.links))
    end = time.time()
    print("#" + str(i), "Time:", end - start)
```
```
1
#0 Time: 0.003222942352294922
1
#1 Time: 0.07713627815246582
1
#2 Time: 0.20673131942749023
1
#3 Time: 0.4618215560913086
1
#4 Time: 1.0132341384887695
1
#5 Time: 2.109161376953125
1
#6 Time: 4.251365661621094
1
#7 Time: 8.811024188995361
1
#8 Time: 17.281348705291748
```

Python 3.6
MaaS  2.5.2 (7523-ge4ecbd54d-0ubuntu1~18.04.1)

**Comment by JeremyMordkoff** [comment]
I have observed this also -- Ubuntu 16, MAAS 2.5.3

**Comment by blakerouse** [comment]
Fixed in #216
